### PR TITLE
Add ipv4/ipv6 proxy, avoid connecting to invalid IPs

### PR DESCRIPTION
**Pull request by jonasschnelli**
This will add two new startup arguments (`-i` and `-k`) for setting an IPv4 or/and an IPv6 proxy.

### CODE REVIEW

**Comment by laanwj** [comment]
Nice.
It may be slightly more consistent to add a IPv6 proxy option as well at the same time - if it's easy to do.

**Comment by Magicking** [comment]
Shouldn't it be labelled as SOCKS proxy instead of IPv4 proxy and allow to set the version of the SOCKS protocol used ?

**Comment by jonasschnelli** [comment]
Also added support for an additional IPv6 proxy.

**Comment by laanwj** [comment]
@Magicking We only support SOCKS5 over here. I suppose the help message needs to mention that, though `SOCKS5 proxy to use for IPv4` etc.

File: main.cpp [hunk 1]
```diff
@@ -58,6 +62,8 @@ class CDnsSeedOpts {
         {"dnsthreads", required_argument, 0, 'd'},
         {"port", required_argument, 0, 'p'},
         {"onion", required_argument, 0, 'o'},
+        {"proxyipv4", required_argument, 0, 'i'},
+        {"proxyipv6", required_argument, 0, 'ik'},
```
Comment (laanwj) [inline main.cpp]: 'ik' looks wrong

Comment (jonasschnelli) [inline main.cpp]: Oops. Last minut type. Will fix

**Comment by jonasschnelli** [comment]
It would be possible to support Socks4 Proxy by just passing a '4' as third param for SetProxy(), but id like not to ass more complexity to the arguments setting. If one needs Socks4 support he should be able to compile this by himself. The seeder has a rar user base.

**Comment by jonasschnelli** [comment]
Fix wrong `ik` argument mapping and also mention SOCKS5 protocol in the help/usage text.

**Comment by laanwj** [comment]
No one should be using SOCKS4 anymore in this day and age.

**Comment by sipa** [comment]
ACK

**Comment by schildbach** [comment]
Thanks for working on this!

**Pull request merged**